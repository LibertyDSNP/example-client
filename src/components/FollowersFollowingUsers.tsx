import React from "react";
import UserAvatar from "./UserAvatar";
import { Button } from "antd";

interface FollowersFollowingUsersProps {
  selectedListTitle: string;
  tempUserList: any[];
}

const FollowersFollowingUsers = ({
  selectedListTitle,
  tempUserList,
}: FollowersFollowingUsersProps): JSX.Element => {
  const userList = () => {
    switch (selectedListTitle) {
      case "showFollowers":
        return tempUserList.filter((user) => user.followsMe);
      case "showFollowing":
        return tempUserList.filter((user) => user.following);
      default:
        return [];
    }
  };

  return (
    <>
      {userList().map((user, index) => (
        <div className="FollowersFollowingUsers__user" key={index}>
          <UserAvatar avatarSize="small" profile={user.profile} />
          <div className="FollowersFollowingUsers__name">{user.name}</div>
          <Button
            className="FollowersFollowingUsers__button"
            name={user.following ? "Unfollow" : "Follow"}
          >
            {user.following ? "Unfollow" : "Follow"}
          </Button>
        </div>
      ))}
    </>
  );
};
export default FollowersFollowingUsers;
