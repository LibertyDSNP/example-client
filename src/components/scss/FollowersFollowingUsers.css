/* Error: Undefined variable.
 *    ,
 * 30 |     color: $black;
 *    |            ^^^^^^
 *    '
 *   FollowersFollowingUsers.scss 30:12  root stylesheet */

body::before {
  font-family: "Source Code Pro", "SF Mono", Monaco, Inconsolata, "Fira Mono",
      "Droid Sans Mono", monospace, monospace;
  white-space: pre;
  display: block;
  padding: 1em;
  margin-bottom: 1em;
  border-bottom: 2px solid black;
  content: "Error: Undefined variable.\a    \2577 \a 30 \2502      color: $black;\a    \2502             ^^^^^^\a    \2575 \a   FollowersFollowingUsers.scss 30:12  root stylesheet";
}
