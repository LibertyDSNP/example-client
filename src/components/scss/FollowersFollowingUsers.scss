.FollowersFollowingUsers__user {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 6px;
}

.FollowersFollowingUsers__name {
  flex: 1;
  padding-left: 16px;
}

.FollowersFollowingUsers__button {
  border-radius: 6px;
}

.FollowersFollowingUsers__buttonFollowIcon {
  transform: rotate(0deg);
  transition: transform 0.3s;
}

.FollowersFollowingUsers__buttonUnfollowIcon {
  transform: rotate(45deg);
}

.FollowersFollowingUsers__button[name="Unfollow"] {
  &:hover,
  &:active,
  &:focus {
    color: $black;
    border-color: $green;
    background-color: $light-gray;
  }
}

.FollowersFollowingUsers__button[name="Follow"] {
  background-color: $black;
  color: $white;

  &:hover,
  &:active,
  &:focus {
    color: $white;
    border-color: $green;
    background-color: $dark-gray;
  }
}
