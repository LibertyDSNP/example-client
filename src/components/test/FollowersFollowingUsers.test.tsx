import React from "react";
import FollowersFollowingUsers from "../FollowersFollowingUsers";
import { shallow, mount } from "enzyme";
import { preFabProfiles } from "../../test/testProfiles";

const mockTempUserList = [
  {
    profile: preFabProfiles[0],
    icon: preFabProfiles[0].icon,
    name: preFabProfiles[0].name,
    following: true,
    followsMe: true,
  },
  {
    profile: preFabProfiles[1],
    icon: preFabProfiles[1].icon,
    name: preFabProfiles[1].name,
    following: true,
    followsMe: false,
  },
  {
    profile: preFabProfiles[2],
    icon: preFabProfiles[2].icon,
    name: preFabProfiles[2].name,
    following: false,
    followsMe: true,
  },
];

describe("Feed", () => {
  it("renders without crashing", () => {
    expect(() => {
      shallow(
        <FollowersFollowingUsers
          selectedListTitle="showFollowers"
          tempUserList={mockTempUserList}
        />
      );
    }).not.toThrow();
  });
  it("to match snapshot", () => {
    const component = mount(
      <FollowersFollowingUsers
        selectedListTitle="showFollowers"
        tempUserList={mockTempUserList}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
