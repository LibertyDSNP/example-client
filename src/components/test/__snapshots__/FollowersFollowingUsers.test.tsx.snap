// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed to match snapshot 1`] = `
<FollowersFollowingUsers
  selectedListTitle="showFollowers"
  tempUserList={
    Array [
      Object {
        "following": true,
        "followsMe": true,
        "icon": Object {
          "url": "https://image.shutterstock.com/image-vector/monday-time-sparkle-shine-word-600w-731826949.jpg",
        },
        "name": "Monday January",
        "profile": Object {
          "@context": "https://www.w3.org/ns/activitystreams",
          "actor": "0x1DCODE0000",
          "discoverable": true,
          "icon": Object {
            "url": "https://image.shutterstock.com/image-vector/monday-time-sparkle-shine-word-600w-731826949.jpg",
          },
          "id": "http://localhost:3003/api/announce/0x1DCODE0000",
          "name": "Monday January",
          "preferredUsername": "0Monday0",
          "socialAddress": "0x1DCODE0000",
          "summary": "",
          "type": "Person",
          "url": "",
          "walletAddress": "0xCODE0000",
        },
      },
      Object {
        "following": true,
        "followsMe": false,
        "icon": Object {
          "url": "https://image.shutterstock.com/image-vector/traditional-taco-tuesday-neon-light-600w-1193206603.jpg",
        },
        "name": "Tuesday February",
        "profile": Object {
          "@context": "https://www.w3.org/ns/activitystreams",
          "actor": "0x1DCODE1111",
          "discoverable": true,
          "icon": Object {
            "url": "https://image.shutterstock.com/image-vector/traditional-taco-tuesday-neon-light-600w-1193206603.jpg",
          },
          "id": "http://localhost:3003/api/announce/0x1DCODE1111",
          "name": "Tuesday February",
          "preferredUsername": "1Tuesday1",
          "socialAddress": "0x1DCODE1111",
          "summary": "",
          "type": "Person",
          "url": "",
          "walletAddress": "0xCODE1111",
        },
      },
      Object {
        "following": false,
        "followsMe": true,
        "icon": Object {
          "url": "https://image.shutterstock.com/image-vector/wednesday-pop-art-illustration-vector-600w-219333010.jpg",
        },
        "name": "Wednesday March",
        "profile": Object {
          "@context": "https://www.w3.org/ns/activitystreams",
          "actor": "0x1DCODE2222",
          "discoverable": true,
          "icon": Object {
            "url": "https://image.shutterstock.com/image-vector/wednesday-pop-art-illustration-vector-600w-219333010.jpg",
          },
          "id": "http://localhost:3003/api/announce/0x1DCODE2222",
          "name": "Wednesday March",
          "preferredUsername": "2Wednesday2",
          "socialAddress": "0x1DCODE2222",
          "summary": "",
          "type": "Person",
          "url": "",
          "walletAddress": "0xCODE2222",
        },
      },
    ]
  }
>
  <div
    className="FollowersFollowingUsers__user"
    key="0"
  >
    <UserAvatar
      avatarSize="small"
      profile={
        Object {
          "@context": "https://www.w3.org/ns/activitystreams",
          "actor": "0x1DCODE0000",
          "discoverable": true,
          "icon": Object {
            "url": "https://image.shutterstock.com/image-vector/monday-time-sparkle-shine-word-600w-731826949.jpg",
          },
          "id": "http://localhost:3003/api/announce/0x1DCODE0000",
          "name": "Monday January",
          "preferredUsername": "0Monday0",
          "socialAddress": "0x1DCODE0000",
          "summary": "",
          "type": "Person",
          "url": "",
          "walletAddress": "0xCODE0000",
        }
      }
    >
      <Avatar
        alt="Monday January"
        shape="circle"
        size={32}
        src="data:image/png;base64,00"
      >
        <span
          className="ant-avatar ant-avatar-circle ant-avatar-image"
          style={
            Object {
              "fontSize": 18,
              "height": 32,
              "lineHeight": "32px",
              "width": 32,
            }
          }
        >
          <img
            alt="Monday January"
            onError={[Function]}
            src="data:image/png;base64,00"
          />
        </span>
      </Avatar>
    </UserAvatar>
    <div
      className="FollowersFollowingUsers__name"
    >
      Monday January
    </div>
    <Button
      className="FollowersFollowingUsers__button"
      name="Unfollow"
    >
      <Wave>
        <button
          className="ant-btn FollowersFollowingUsers__button"
          name="Unfollow"
          onClick={[Function]}
          type="button"
        >
          <LoadingIcon
            existIcon={false}
            loading={false}
            prefixCls="ant-btn"
          >
            <CSSMotion
              motionName="ant-btn-loading-icon-motion"
              onAppearActive={[Function]}
              onAppearStart={[Function]}
              onEnterActive={[Function]}
              onEnterStart={[Function]}
              onLeaveActive={[Function]}
              onLeaveStart={[Function]}
              removeOnLeave={true}
              visible={false}
            >
              <DomWrapper />
            </CSSMotion>
          </LoadingIcon>
          <span
            key=".0"
          >
            Unfollow
          </span>
        </button>
      </Wave>
    </Button>
  </div>
  <div
    className="FollowersFollowingUsers__user"
    key="1"
  >
    <UserAvatar
      avatarSize="small"
      profile={
        Object {
          "@context": "https://www.w3.org/ns/activitystreams",
          "actor": "0x1DCODE2222",
          "discoverable": true,
          "icon": Object {
            "url": "https://image.shutterstock.com/image-vector/wednesday-pop-art-illustration-vector-600w-219333010.jpg",
          },
          "id": "http://localhost:3003/api/announce/0x1DCODE2222",
          "name": "Wednesday March",
          "preferredUsername": "2Wednesday2",
          "socialAddress": "0x1DCODE2222",
          "summary": "",
          "type": "Person",
          "url": "",
          "walletAddress": "0xCODE2222",
        }
      }
    >
      <Avatar
        alt="Wednesday March"
        shape="circle"
        size={32}
        src="data:image/png;base64,00"
      >
        <span
          className="ant-avatar ant-avatar-circle ant-avatar-image"
          style={
            Object {
              "fontSize": 18,
              "height": 32,
              "lineHeight": "32px",
              "width": 32,
            }
          }
        >
          <img
            alt="Wednesday March"
            onError={[Function]}
            src="data:image/png;base64,00"
          />
        </span>
      </Avatar>
    </UserAvatar>
    <div
      className="FollowersFollowingUsers__name"
    >
      Wednesday March
    </div>
    <Button
      className="FollowersFollowingUsers__button"
      name="Follow"
    >
      <Wave>
        <button
          className="ant-btn FollowersFollowingUsers__button"
          name="Follow"
          onClick={[Function]}
          type="button"
        >
          <LoadingIcon
            existIcon={false}
            loading={false}
            prefixCls="ant-btn"
          >
            <CSSMotion
              motionName="ant-btn-loading-icon-motion"
              onAppearActive={[Function]}
              onAppearStart={[Function]}
              onEnterActive={[Function]}
              onEnterStart={[Function]}
              onLeaveActive={[Function]}
              onLeaveStart={[Function]}
              removeOnLeave={true}
              visible={false}
            >
              <DomWrapper />
            </CSSMotion>
          </LoadingIcon>
          <span
            key=".0"
          >
            Follow
          </span>
        </button>
      </Wave>
    </Button>
  </div>
</FollowersFollowingUsers>
`;
